src/fides/core/export.py

```diff
@@ -317,15 +317,11 @@ def generate_system_records(  # pylint: disable=too-many-nested-blocks, too-many
                     for dataset_reference in dataset_references
                 ]
                 cartesian_product_of_declaration = []
-                if system_custom_field_data:
-                    for _, v in system_custom_field_data.items():
-                        for product in cartesian_product_of_declaration_builder:
+                for product in cartesian_product_of_declaration_builder:
+                    if system_custom_field_data:
+                        for _, v in system_custom_field_data.items():
                             product.append(v)
-                            cartesian_product_of_declaration.append(tuple(product))
-                else:
-                    cartesian_product_of_declaration = [
-                        tuple(x) for x in cartesian_product_of_declaration_builder
-                    ]
+                    cartesian_product_of_declaration.append(tuple(product))
```

Comment:
So the real _change_ here seems to be basically inverting the order of this, i.e.
* Before: for each custom field and each product....
* After: for each product and each custom field...

Can you provide some explanation as to why this fixes the issue?

Reply:
right, but with one more key detail: we move the `cartesian_product_of_declaration.append(tuple(product))` operation into the level of _for each product_  - which is only really possible if we invert the nesting order as you mention.

i tried to summarize this in the PR description with this point:
> for systems with privacy declarations and with custom field data, add in all of the custom field data per output row, rather than adding new output rows for each custom field

but maybe that's keeping it too high-level? i can add in some more detail to hopefully make that clearer

Reply:
@NevilleS i've updated the PR description to explain the fix a bit more thoroughly. let me know what you think - i hope i haven't made things more confusing!